tests/resource/test_resource.py

```diff
@@ -180,6 +191,185 @@ def test_ipython_display(self, r, rep, capsys):
         captured = capsys.readouterr()
         assert rep in captured.out
 
+    expected_results_add_series = (
+        Resources(
+            2,
+            6,
+            defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 1}),
+            defaultdict(int, {1: 5, 2: 1}),
+        ),
+        Resources(
+            5,
+            6,
+            defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 1}),
+            defaultdict(int, {1: 5, 2: 1}),
+        ),
+        Resources(
+            2,
+            9,
+            defaultdict(int, {"RZ": 2, "CNOT": 1, "RY": 2, "Hadamard": 2, "PauliZ": 2}),
+            defaultdict(int, {1: 8, 2: 1}),
+        ),
+        Resources(
+            4,
+            8,
+            defaultdict(int, {"RZ": 2, "CNOT": 2, "RY": 2, "Hadamard": 2}),
+            defaultdict(int, {1: 6, 2: 2}),
+        ),
+    )
+
+    @pytest.mark.parametrize(
+        "resource_obj, expected_res_obj", zip(resource_quantities, expected_results_add_series)
+    )
+    def test_add_in_series(self, resource_obj, expected_res_obj):
+        """Test the add_in_series function works with Resoruces"""
+        resource_obj = copy.deepcopy(resource_obj)
```

Comment:
Do we need the deep copy here? There is no inplace modifications

Reply:
It's not needed, it was just leftover from when I copied the tests in labs.